app/javascript/packs/AddNewRestPopup.jsx

```diff
@@ -0,0 +1,151 @@
+import React from 'react';
+import $ from 'jquery';
+
+class AddNewRestPopup extends React.Component {
+
+    constructor() {
+        super();
+        this.state = {
```

Comment:
With babel you can put your initialized state outside the ctor: https://michalzalecki.com/react-components-and-class-properties/